src/cli/rustup_mode.rs

```diff
@@ -1578,6 +1585,11 @@ fn set_profile(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
     Ok(utils::ExitCode(0))
 }
 
+fn set_auto_self_update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
+    cfg.set_auto_self_update(&m.value_of("auto-self-update-mode").unwrap())?;
+    Ok(utils::ExitCode(0))
+}
+
```

Comment:
I wonder if this function should warn the user if rustup was built with the `no-self-update` feature and they set it to `enable` ?

Reply:
Added.